File: backend/appointment-service/src/main/java/com/mayur/appointment/domain/Appointment.java
package com.mayur.appointment.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String patientId; // MRN or Patient ID from Patient Service

    @Column(nullable = false)
    private Long doctorId; // ID of the doctor (Internal User ID)

    @Column(nullable = false)
    private LocalDateTime appointmentTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AppointmentStatus status;
    
    @Column(columnDefinition = "TEXT")
    private String reason;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public enum AppointmentStatus {
        SCHEDULED,
        CHECKED_IN,
        IN_PROGRESS,
        COMPLETED,
        CANCELLED,
        NO_SHOW
    }
}

File: frontend/src/types/appointment.ts
export type AppointmentStatus = 'SCHEDULED' | 'CONFIRMED' | 'COMPLETED' | 'CANCELLED';

export interface Appointment {
    id: number;
    patientId: string; // MRN
    doctorId: number;
    reason: string;
    appointmentTime: string; // ISO 8601 LocalDateTime
    status: AppointmentStatus;
}

export interface CreateAppointmentRequest {
    patientId: string;
    doctorId: number;
    appointmentTime: string; // ISO 8601 LocalDateTime
    reason: string;
}

// Placeholder for Doctor type until we have a real Doctor Service or use Identity
export interface Doctor {
    id: number;
    name: string;
    specialization: string;
}
